test/TestCaseReader.cpp

```diff
@@ -137,6 +146,56 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSource = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSource;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSource.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSource.substr(0, remappingPos));
+					externalSource = boost::trim_copy(externalSource.substr(remappingPos + 1));
+				}
+
+				fs::path testCaseParentDir = canonical(fs::path{m_fileName}).parent_path();
```

Comment:
I think there is an implicit conversion from string:
```suggestion
				fs::path testCaseParentDir = canonical(m_fileName).parent_path();
```

Reply:
And now `m_fileName` is `fs::path`.